sound/soc/sof/ipc4-loader.c

```diff
@@ -423,6 +423,65 @@ int sof_ipc4_reload_fw_libraries(struct snd_sof_dev *sdev)
 	return ret;
 }
 
+/**
+ * sof_ipc4_update_cpc_from_manifest - Update the cpc in base config from manifest
+ * @sdev: SOF device
+ * @fw_module: pointer struct sof_ipc4_fw_module to parse
+ * @basecfg: Pointer to the base_config to update
+ */
+void sof_ipc4_update_cpc_from_manifest(struct snd_sof_dev *sdev,
+				       struct sof_ipc4_fw_module *fw_module,
+				       struct sof_ipc4_base_module_cfg *basecfg)
+{
+	const struct sof_man4_module_config *fw_mod_cfg;
+	u32 cpc_pick = 0;
+	u32 max_cpc = 0;
+	int i;
+
+	if (!fw_module || !fw_module->fw_mod_cfg)
+		return;
+
+	/*
+	 * Find the best matching (highest) CPC value based on the module's
+	 * external configuration (IBS/OBS).
```

Comment:
what did you mean by 'external'? Did you mean "the module IBS/OBS configuration retrieved from the topology"?

Reply:
The IBS/OBS is in mod_cfg along with the CPC as triplet.
The IBS/OBS for the used configuration is coming from topology when the module is configured for the stream.
IBS/OBS is the 'external' configuration in a sense that it is the input and output buffer size, which is related to the audio formats picked for the given side of the module.

Reply:
@ujfalusi I am in agreement with your explanation, but still 'external' is very ambiguous.

Would this work for you

"
Find the best matching (highest) CPC value based on the module's IBS/OBS configuration inferred from the audio format selection.
"

Reply:
@lgirdwood, @plbossart, yes, I forgot to update this, in the next version I will correct.